server/controllers/categoryController.js

```diff
@@ -257,34 +259,66 @@ class CategoryController {
    */
   static getAllArticlesForACategory(req, res, next) {
     const { categoryName } = req.params;
+    const { query } = req;
+    const limit = Number(query.limit) || 6;
+    const currentPage = Number(query.page) || 1;
+    const offset = (currentPage - 1) * limit;
+
     Category.findOne({
       where: { name: categoryName },
       include: [
         {
           model: Article,
-          as: 'category',
+          as: 'articles',
           attributes: { exclude: ['body', 'authorId', 'displayStatus', 'updatedAt'] },
-          include: [
-            {
-              model: User,
-              as: 'author',
-              attributes: ['firstName', 'lastName', 'email', 'image'],
-            },
-          ],
         },
       ],
     })
       .then((category) => {
         if (!category) {
           return res.status(404).json({
             status: 'error',
-            error: 'There are no articles in this category'
+            error: 'Category does not exist'
           });
         }
-        return res.status(200).json({
-          status: 'success',
-          category
-        });
+        const articleIds = category.articles.map(article => article.id);
+        Article.findAndCountAll({
+          where: { id: articleIds },
+          include: [
+            {
+              model: User,
+              as: 'author',
+              attributes: ['firstName', 'lastName', 'email', 'image'],
+            },
+            {
+              model: Tag,
+              as: 'tags',
+              attributes: ['tag'],
+              through: {
+                attributes: [],
+              },
+            }
+          ],
+          attributes: ['id', 'slug', 'title', 'timeToRead', 'description', 'createdAt', 'updatedAt', 'authorId'],
+          limit,
+          offset
+        })
+          .then((articles) => {
+            if (articles.rows.length === 0) {
+              return res.status(404).json({
+                status: 'error',
+                error: 'There are no articles in this category'
+              });
+            }
+            const pagination = paginateArticle(articles, currentPage, limit);
```

Comment:
I don't think it should return an error if no articles exist for the category. It should return an empty array instead.

Reply:
I have fixed it

Reply:
Thanks for the feedback. I have fixed it. It now returns an empty array for categories without articles.

Reply:
Thanks for the feedback. I have implemented the feedback.

Reply:
Thanks for the feedback. I have implemented the feedback.

Reply:
Thanks for the feedback. I have implemented the feedback.

Reply:
Thanks for the feedback. I have implemented the feedback.

Reply:
Thanks for the feedback. I have implemented the feedback.